elasticsearch/src/test/java/org/hibernate/search/elasticsearch/test/DefaultElasticsearchClientFactoryTest.java

```diff
@@ -331,12 +331,18 @@ public void discovery() throws Exception {
 			Response result = doPost( client, "/myIndex/myType", payload );
 			assertThat( result.getStatusLine().getStatusCode() ).as( "status code" ).isEqualTo( 200 );
 
-			Thread.sleep( 2000 ); // Wait for the refresh to occur
+			Thread.sleep( 4000 ); // Wait for the refresh to occur
```

Comment:
Needing this test to wait `4s` is annoying and I would accept that if it was the only way, but we all know that these figures are going to fail in some other slightly different condition.

Is there no way with this mock library to "wait until" the condition is satisfied?

I'm confident you could code that yourself but I'd also expect a library like this to have some out of the box support for such kind of tests?